src/key.rs

```diff
@@ -74,6 +74,15 @@ impl RedisKey {
         };
         Ok(val)
     }
+
+    pub fn hash_get(&self, field: &str) -> Result<Option<RedisString>, RedisError> {
+        let val = if self.is_null() {
```

Comment:
Isn't this unnecessary since `hash_get_key` already does null detection? Or is it meant to improve performance by avoiding the need to open the key?

Reply:
good question, I followed the same pattern you did in line 70 with read()